com.woltlab.wcf/templates/userNotice.tpl

```diff
@@ -10,10 +10,21 @@
 		<p class="info" role="status">{lang}wcf.page.availableUpdates{/lang}</p>
 	{/if}
 	
-	{if $__wcf->user->activationCode && REGISTER_ACTIVATION_METHOD == 1 && $templateName != 'registerActivation' && $templateName != 'register' && $templateName != 'redirect' && $__wcf->user->getBlacklistMatches()|empty}
+	{* user needs email confirmation to be activated *}
+	{if !$__wcf->user->isEmailConfirmed() && !$__wcf->user->isActivated() && REGISTER_ACTIVATION_METHOD & 1 && $templateName != 'registerActivation' && $templateName != 'register' && $templateName != 'redirect' && $__wcf->user->getBlacklistMatches()|empty}
```

Comment:
Refactor this to move `$templateName != 'registerActivation' && $templateName != 'register' && $templateName != 'redirect' && $__wcf->user->getBlacklistMatches()|empty` into a dedicated top-level `{if}`.